### PR TITLE
Implement i128 and u128.

### LINKED ISSUE #199: impl Format for  i128/u128
Would also be nice to have i128/u128 while at it.

_Originally posted by @MathiasKoch in https://github.com/knurling-rs/defmt/issues/160#issuecomment-687833958_

### PR DESCRIPTION
**Pull request by BriocheBerlin**
Closes #199

### CODE REVIEW

File: firmware/qemu/src/bin/log.rs [hunk 1]
```diff
@@ -419,6 +419,22 @@ fn main() -> ! {
...
 
+    defmt::info!(
+        "i128: 0 = {:i128}, -1 = {:i128}, MAX = {:i128}, MIN = {:i128}",
+        0,
+        -1,
+        i128::max_value(),
+        i128::min_value()
+    );
+
+    defmt::info!(
```
Comment (japaric) [inline firmware/qemu/src/bin/log.rs:430]: could you also test here that we can format `u128` and `i128` values using the `{:?}` format specifier?

**Review by japaric — APPROVED** [review]
Looks great! Extending the format string is a breaking change so this needs to wait until v0.2.0 but I'll mark this as approved.

File: firmware/qemu/src/bin/log.rs (hunk 1 above)
Comment (BriocheBerlin) [reply firmware/qemu/src/bin/log.rs:430]: Is it ok now?

Comment (japaric) [reply firmware/qemu/src/bin/log.rs:430]: yes, looks good; thanks! I'll re-approve the PR but it will need a rebase before it can be merged.

**Review by japaric — APPROVED** [review]

**Comment by Lotterleben** [comment]
Awesome, feel free to squash and merge!

**Pull request merged**